src/classes/UTIL_DMLService.cls

```diff
@@ -0,0 +1,298 @@
+    * @return List<Database.SaveResult>
+    **********************************************************************************************************/
+    public static List<Database.SaveResult> updateRecords(List<sObject> recordsToUpdate, Boolean allOrNone) {
+        Database.DMLOptions dmlOptions = new Database.DMLOptions();
+        
+        dmlOptions.optAllOrNone = allOrNone;
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+
+        return Database.update(recordsToUpdate, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class update method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordToUpdate The sObject to update in the org.
+    * @param dmlOptions The optional parameter that specifies additional data for the transaction.
+    * @return Database.SaveResult
+    **********************************************************************************************************/
+    public static Database.SaveResult updateRecord(sObject recordToUpdate, Database.DMLOptions dmlOptions) {
+        if (!useAllowSaveProperty(dmlOptions)) {
+            dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+        }
+
+        return Database.update(recordToUpdate, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class update method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordsToUpdate The sObjects to update in the org.
+    * @param dmlOptions The optional parameter that specifies additional data for the transaction.
+    * @return List<Database.SaveResult>
+    **********************************************************************************************************/
+    public static List<Database.SaveResult> updateRecords(List<sObject> recordsToUpdate, Database.DMLOptions dmlOptions) {
+        if (!useAllowSaveProperty(dmlOptions)) {
+            dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+        }
+
+        return Database.update(recordsToUpdate, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class upsert method.
+    * @param recordToUpsert The sObject to upsert in the org.
+    * @return Database.UpsertResult
+    **********************************************************************************************************/
+    public static Database.UpsertResult upsertRecord(sObject recordToUpsert) {
+        return Database.upsert(recordToUpsert);
```

Comment:
don't your new upsert methods need to set the allowSave property just like the insert and update covers?

Reply:
Database.upsert(...) does not support Database.DMLOptions. This basically wraps the upsert method for consistency.